source/postprocess/visualization/named_additional_outputs.cc

```diff
@@ -0,0 +1,188 @@
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/named_additional_outputs.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/utilities.h>
+
+#include <algorithm>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      NamedAdditionalOutputs<dim>::
+      NamedAdditionalOutputs ()
+        :
+        DataPostprocessor<dim> ()
+      {}
+
+
+      template <int dim>
+      void
+      NamedAdditionalOutputs<dim>::
+      initialize ()
+      {
+        MaterialModel::MaterialModelOutputs<dim> out(0,
+                                                     this->n_compositional_fields());
+        this->get_material_model().create_additional_named_outputs(out);
+
+        for (unsigned int k=0; k<out.additional_outputs.size(); ++k)
+          {
+            const MaterialModel::NamedAdditionalMaterialOutputs<dim> *result
+              = dynamic_cast<const MaterialModel::NamedAdditionalMaterialOutputs<dim> *> (out.additional_outputs[k].get());
+
+            if (result)
+              {
+                std::vector<std::string> names = result->get_names();
+
+                for (unsigned int i=0; i<names.size(); ++i)
+                  property_names.push_back(names[i]);
+              }
+          }
+
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          std::replace(property_names[i].begin(),property_names[i].end(),' ', '_');
+      }
+
+
+
+      template <int dim>
+      std::vector<std::string>
+      NamedAdditionalOutputs<dim>::
+      get_names () const
+      {
+        return property_names;
+      }
+
+
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      NamedAdditionalOutputs<dim>::
+      get_data_component_interpretation () const
+      {
+        return std::vector<DataComponentInterpretation::DataComponentInterpretation> (get_names().size(),
+                                                                                      DataComponentInterpretation::component_is_scalar);
+      }
+
+
+
+      template <int dim>
+      UpdateFlags
+      NamedAdditionalOutputs<dim>::
+      get_needed_update_flags () const
+      {
+        return update_gradients | update_values  | update_q_points;
+      }
+
+
+
+      template <int dim>
+      void
+      NamedAdditionalOutputs<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,           ExcInternalError());
```

Comment:
break the exception class to the next line; same for the previous line with the awkward spacing